File: py/IEC62325/MarketOperations/MarketSystem/ExternalInputs/DefaultBidCurveData.py
from IEC61970.Base.Core.CurveData import CurveData
from IEC62325.MarketOperations.MktDomain.BidCalculationBasis import BidCalculationBasis


class DefaultBidCurveData(CurveData):
    #  * Curve data for default bid curve and startup cost curve.
    #  * @created 27-Dec-2023 3:27:05 PM

    def __init__(self) -> None:
        super().__init__()
        # 	 * Type of calculation basis used to define the default bid segment curve.
        self.bid_segment_calc_type: BidCalculationBasis = BidCalculationBasis()

File: java/IEC62325/MarketOperations/MarketSystem/ExternalInputs/DefaultBidCurveData.java
package IEC62325.MarketOperations.MarketSystem.ExternalInputs;

import IEC62325.MarketOperations.MktDomain.BidCalculationBasis;
import IEC61970.Base.Core.CurveData;

/**
 * Curve data for default bid curve and startup cost curve.
 * @created 03-Jan-2024 2:08:10 PM
 */
public class DefaultBidCurveData extends CurveData {

	/**
	 * Type of calculation basis used to define the default bid segment curve.
	 */
	public BidCalculationBasis bidSegmentCalcType;

	public DefaultBidCurveData(){

	}
}//end DefaultBidCurveData